h2/src/main/org/h2/fulltext/FullTextSettings.java

```diff
@@ -10,20 +10,22 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
-import java.util.HashMap;
-import java.util.HashSet;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
 import org.h2.util.New;
 import org.h2.util.SoftHashMap;
 
 /**
  * The global settings of a full text search.
  */
-class FullTextSettings {
+final class FullTextSettings {
 
     /**
      * The settings of open indexes.
      */
-    private static final HashMap<String, FullTextSettings> SETTINGS = New.hashMap();
+    private static final Map<String, FullTextSettings> SETTINGS = Collections.synchronizedMap(New.<String, FullTextSettings>hashMap());
 
```

Comment:
why do all these need to be synch collections?
 In general I do not think synch collections ever make much sense. Because you can't synch properly when doing iteration over them, and people never remember to synch across read/write pairs, etc, so you might just as well declare some other object as the lock and synch on that object.
Sorry if this doesn't make too much sense, rather late here :-)